From 0606a55bf7db0d894ff3492aacbd45faee972383 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Fri, 12 May 2017 21:16:47 +0800
Subject: [PATCH 1780/5242] MLK-14893 ARM64: dts: freescale: imx8qm: add CA72
 only dtb

commit  e2897fea5d553c2fca6348784977c4547799db59 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add CA72 only dtb to support booting up kernel with
CA72 cluster only, need to use CA72 specific flash.bin
and CA72 specific dtb to boot up kernel.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/Makefile             |    2 +-
 .../freescale/fsl-imx8qm-lpddr4-arm2_ca72_only.dts |   32 ++++++++++++++++++++
 2 files changed, 33 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2_ca72_only.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 7523e63..2d196e6 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -13,4 +13,4 @@ dtb-$(CONFIG_ARCH_LAYERSCAPE) += fsl-ls2080a-rdb.dtb
 dtb-$(CONFIG_ARCH_LAYERSCAPE) += fsl-ls2080a-simu.dtb
 dtb-$(CONFIG_ARCH_LAYERSCAPE) += fsl-ls2088a-qds.dtb
 dtb-$(CONFIG_ARCH_LAYERSCAPE) += fsl-ls2088a-rdb.dtb
-dtb-$(CONFIG_ARCH_FSL_IMX8QM) += fsl-imx8qm-lpddr4-arm2.dtb fsl-imx8qm-lpddr4-arm2_ca72.dtb
+dtb-$(CONFIG_ARCH_FSL_IMX8QM) += fsl-imx8qm-lpddr4-arm2.dtb fsl-imx8qm-lpddr4-arm2_ca72.dtb fsl-imx8qm-lpddr4-arm2_ca72_only.dtb
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2_ca72_only.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2_ca72_only.dts
new file mode 100644
index 0000000..a3011d2
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2_ca72_only.dts
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2016 Freescale Semiconductor, Inc.
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8qm-lpddr4-arm2_ca72.dts"
+
+&A53_0 {
+	device_type = "";
+};
+
+&A53_1 {
+	device_type = "";
+};
+
+&A53_2 {
+	device_type = "";
+};
+
+&A53_3 {
+	device_type = "";
+};
-- 
1.7.9.5

